gitdiff/apply.go

```diff
@@ -259,6 +259,16 @@ func (a *Applier) ApplyTextFragment(dst io.Writer, f *TextFragment) error {
 		}
 	}
 	a.nextLine = fragStart + used
+
+	// new position of +0,0 mean a full delete, so check for leftovers
+	if f.NewPosition == 0 && f.NewLines == 0 {
+		var b [1][]byte
+		n, err := a.lineSrc.ReadLinesAt(b[:], a.nextLine)
+		if err != io.EOF || n > 0 {
+			return applyError(&Conflict{"src still has content after full delete"}, lineNum(a.nextLine))
+		}
```

Comment:
If there's a non-`EOF` error, I should wrap that instead of swallowing it as a conflict. This is also similar to the `isLen` function, but that only works for plain `io.ReaderAt` not `LineReaderAt`.